DLM frozen tier docs (#6530)
elastic/docs-content: manage-data/lifecycle/data-stream.md

# Data stream lifecycle in {{es}} [data-stream-lifecycle]

> A data stream lifecycle in {{es}} is a built-in automation mechanism for managing data retention, performance, storage cost, and ingest efficiency.

Reviewer: mostly out of curiosity: how does this improve ingest efficiency?

Author: Good question. I don't remember knowingly making that change so I'll blame my robot friend and revert it in 52e33b10430b958412229633af37e264b9818387

> By configuring rollover, retention, downsampling, and frozen phase transitions to long-term searchable archives, your [data streams](/manage-data/data-store/data-streams.md) stay efficient as they age.

Reviewer 2: Again to avoid the word "phase":

```suggestion
By configuring rollover, retention, downsampling, and frozen transitions to long-term searchable archives, your [data streams](/manage-data/data-store/data-streams.md) stay efficient as they age.
```

This page explains how the lifecycle works, its key features, and how to configure it for both new and existing data streams.

Data stream lifecycle manages your data streams according to your retention and storage goals. For example, you can configure the lifecycle to:

* Ensure that data indexed in the data stream is kept at least for the retention time you defined.
* Ensure that data older than the retention period is deleted automatically by {{es}} at a later time.

> * {applies_to}`stack: ga 9.5+` Move older backing indices to low-cost searchable storage. Refer to [](/manage-data/lifecycle/data-stream/frozen-searchable-snapshots.md).

Reviewer: this one needs a `serverless: unavailable` tag

Author: Thanks! Fixed in 52e33b10430b958412229633af37e264b9818387

2. Rolls over the write index of the data stream, if it fulfills the conditions defined by [`cluster.lifecycle.default.rollover`](elasticsearch://reference/elasticsearch/configuration-reference/data-stream-lifecycle-settings.md#cluster-lifecycle-default-rollover).
3. After an index is not the write index anymore (that is, the data stream has been rolled over), automatically tail merges the index. Data stream lifecycle executes a merge operation that only targets the long tail of small segments instead of the whole shard. As the segments are organised into tiers of exponential sizes, merging the long tail of small segments is only a fraction of the cost of force merging to a single segment. The small segments would usually hold the most recent data so tail merging will focus the merging resources on the higher-value data that is most likely to keep being queried.
4. If [downsampling]({{es-apis}}operation/operation-indices-put-data-lifecycle) is configured it will execute all the configured downsampling rounds.

> 5. {applies_to}`stack: ga 9.5+` Transitions eligible backing indices to partially mounted {{search-snaps}}. Refer to [](/manage-data/lifecycle/data-stream/frozen-searchable-snapshots.md).

Reviewer: same note about `serverless: unavailable`

Author: Thanks! Fixed in 52e33b10430b958412229633af37e264b9818387

::::{important}
We use the `generation_time` instead of the creation time because this ensures that all data in the backing index have passed the retention period. As a result, the retention period is not the exact time data gets deleted, but the minimum time data will be stored.
::::
::::{note}
Steps `2-5` apply only to backing indices that are not already managed by {{ilm-init}}, meaning that these indices either do not have an {{ilm-init}} policy defined, or if they do, they have [`index.lifecycle.prefer_ilm`](elasticsearch://reference/elasticsearch/configuration-reference/data-stream-lifecycle-settings.md#index-lifecycle-prefer-ilm) set to `false`.
::::

## Configuring data stream lifecycle [data-stream-lifecycle-configuration]

Since the lifecycle is configured on the data stream level, the process to configure a lifecycle on a new data stream and on an existing one differ.

> These tutorials help you set up and manage data streams with data stream lifecycle:

Reviewer: personal pref maybe

```suggestion
The following tutorials help you set up and manage data streams with data stream lifecycle:
```

* To create a new data stream with a lifecycle, add the data stream lifecycle as part of the index template that matches the name of your data stream. Refer to [](data-stream/tutorial-create-data-stream-with-lifecycle.md) for the detailed steps. When a write operation with the name of your data stream reaches {{es}} then the data stream is created with the respective data stream lifecycle.
* To update the lifecycle settings for an individual, existing data stream, use the [data stream lifecycle APIs]({{es-apis}}group/endpoint-data-stream). Refer to [](data-stream/tutorial-update-existing-data-stream.md) for details.
* Retention settings for data streams can be configured both individually, at the data stream level, and globally, for all data streams in a cluster. To learn more, refer to [](/manage-data/lifecycle/data-stream/tutorial-data-stream-retention.md).
* To migrate an existing {{ilm-init}} managed data stream to data stream lifecycle, follow the steps in [](data-stream/tutorial-migrate-ilm-managed-data-stream-to-data-stream-lifecycle.md).

> * {applies_to}`stack: ga 9.5+` To move older backing indices to {{search-snaps}}, refer to [](/manage-data/lifecycle/data-stream/frozen-searchable-snapshots.md).

Reviewer: ```suggestion
* {applies_to}`stack: ga 9.5+` {applies_to}`serverless: unavailable` To move older backing indices to {{search-snaps}}, refer to [](/manage-data/lifecycle/data-stream/frozen-searchable-snapshots.md).
```
